engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -121,7 +124,18 @@ public NadelExecutionStrategy(List<Service> services,
         List<CompletableFuture<OneServiceExecution>> result = new ArrayList<>();
         for (MergedField mergedField : fieldSubSelection.getMergedSelectionSet().getSubFieldsList()) {
             ExecutionStepInfo fieldExecutionStepInfo = executionStepInfoFactory.newExecutionStepInfoForSubField(executionCtx, mergedField, rootExecutionStepInfo);
-            Service service = getServiceForFieldDefinition(fieldExecutionStepInfo.getFieldDefinition());
+
+            boolean usesDynamicService = fieldExecutionStepInfo.getFieldDefinition().getDirectives()
+                    .stream()
+                    .anyMatch(directive -> directive.getName().equals(NadelDirectives.DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.getName()));
+
+            final Service service;
+
+            if(usesDynamicService) {
+                service = serviceExecutionHooks.resolveServiceForField(services, fieldExecutionStepInfo.getField().getName(), fieldExecutionStepInfo.getArguments());
```

Comment:
This needs to be improved. When `service` is `null` here - no service is associated with an ARI format, for instance - the ideal behaviour in my opinion would be to return a GraphQL error for the client for that particular field.
So, if the client queries for 2 nodes, each using a different ARI pattern (one for Jira, another for Bitbucket, for example), and we can only find a mapping for the Jira ARI, I'd like the query to return data for the Jira node and an error for the Bitbucket node.
I couldn't find a mechanism to do that though.
One way of doing it could be by creating a special `Service` object containing a `ServiceExecution` that purposefully returns a result with the error that we want?
If this needs indeed a new mechanism, perhaps we can implement it on a different PR though.

Reply:
I think you can do something like

* ExecutionContext.addError() here.
* Makes a fake ServiceExecution that always returns null

In terms of GraphqlError production - perhaps this hook should return an Either like thing that is a Service or an Error


```
ServiceOrError serviceorError = serviceExecutionHooks.resolveServiceForField()
if (serviceOrError.containsError()) {
   GraphqlError e = serviceOrError.getError()
   excetutionContext.addError(e)
 else {
    service  = serviceOrError.getService()
    ```